### PR TITLE
feat: make port_query_offset accept a number or an array of numbers

### PR DESCRIPTION
**Pull request by podrivo**
Hello!

This PR adds an alternative `port_query_offset`. I'm not sure if this all it needs to add an alternative `port_query_offset`, but it worked on the tests I did.

Currently with an option in the game `V Rising`. Related to this issue (https://github.com/gamedig/node-gamedig/issues/438).

### CODE REVIEW

**Comment by CosminPerRam** [comment]
Heyo, thanks for taking care of this.
I have mixed feelings regarding adding another field option for this -- what if we want to have 3 offsets?
I think making `gameQueryPortOffset` an array would be a better choice, but as the majority of games have only a single offset, it would be great if we could handle this as a `number` or as an `array of numbers`.

File: lib/QueryRunner.js [hunk 1]
```diff
@@ -47,16 +47,34 @@ export default class QueryRunner {
 
     if (userOptions.port) {
       if (!userOptions.givenPortOnly) {
-        if (gameQueryPortOffset) { addAttemptWithPort(userOptions.port + gameQueryPortOffset) }
+
```
Comment (CosminPerRam) [inline lib/QueryRunner.js:50]: Please remove the stray newlines.

File: lib/games.js [hunk 2]
```diff
@@ -2656,7 +2656,7 @@ export const games = {
     name: 'V Rising',
     options: {
       port: 27015,
-      port_query_offset: 1,
+      port_query_offset: [1, 15],
```
Comment (CosminPerRam) [inline lib/games.js:2659]: This should be done in a separate PR to link to that issue.

File: lib/QueryRunner.js [hunk 3]
```diff
@@ -47,16 +47,34 @@ export default class QueryRunner {
...
+        if (gameQueryPortOffset) {
+          if (Array.isArray(gameQueryPortOffset)) {
```
Comment (CosminPerRam) [inline lib/QueryRunner.js:52]: The repeating if's could be resolved be preprocessing `gameQueryPortOffset`.
Before the first if, check if it is an array, if not, make one by the sole value (if any).

Afterwards, its just a matter of `.forEach` it and also the `if (gameQueryPortOffset)` will not be required anymore.

**Review by CosminPerRam — CHANGES_REQUESTED** [review]

File: lib/QueryRunner.js (hunk 1 above)
Comment (podrivo) [reply lib/QueryRunner.js]: Removed!

File: lib/QueryRunner.js (hunk 3 above)
Comment (podrivo) [reply lib/QueryRunner.js]: I think I understood what you meant. Thank you! 🙌 
Just added a new code. Let me know!

File: lib/QueryRunner.js [hunk 4]
```diff
@@ -45,18 +45,25 @@ export default class QueryRunner {
       })
     }
 
+    let portOffsetArray = []
+    if (Array.isArray(gameQueryPortOffset)) {
+      portOffsetArray = gameQueryPortOffset
+    } else if (gameQueryPortOffset) {
+      portOffsetArray.push(gameQueryPortOffset)
+    } else {
+      portOffsetArray.push(0)
```
Comment (CosminPerRam) [inline lib/QueryRunner.js:54]: This is not needed.

File: lib/QueryRunner.js [hunk 5]
```diff
@@ -45,18 +45,25 @@ export default class QueryRunner {
       })
     }
 
+    let portOffsetArray = []
+    if (Array.isArray(gameQueryPortOffset)) {
+      portOffsetArray = gameQueryPortOffset
+    } else if (gameQueryPortOffset) {
+      portOffsetArray.push(gameQueryPortOffset)
```
Comment (CosminPerRam) [inline lib/QueryRunner.js:52]: Even simpler would be:
```js
let portOffsetArray = gameQueryPortOffset
if (!Array.isArray(portOffsetArray)) { // the given offset is not an array apparently, it's a single value, so lets fix that
  portOffsetArray = [gameQueryPortOffset]
}
```
Sorry if I seem to be a little petty about this, just thought it could be better (:

Comment (podrivo) [reply lib/QueryRunner.js:52]: Please, by all means! Your input is very much appreciated!

Comment (podrivo) [reply lib/QueryRunner.js:52]: I did some testing and when we don't have `port_query_offset` present, we get `gameQueryPortOffset` set as `undefined` and this will mess with the attempts for when we don't have a port set by the user.

```js
} else if (gameOptions.port) {
      portOffsetArray.forEach((portOffset) => { addAttemptWithPort(gameOptions.port + portOffset) })
}
```

The line `portOffsetArray.push(0)` was basically setting a `[0]` to the `gameQueryPortOffset`.
So if I remove `port_query_offset` from `games.js` and run this below, I'll get an error.

`node bin/gamedig --type vrising 45.137.247.66`

Comment (podrivo) [reply lib/QueryRunner.js]: I've updated to the code below, so we make sure to always have a `0` to be used when no port is provided.
Let me know if it makes sense! (: 

```js
let portOffsetArray = gameQueryPortOffset
if (!Array.isArray(portOffsetArray)) {
  gameQueryPortOffset ? portOffsetArray = [gameQueryPortOffset] : portOffsetArray = [0]
}
```

Comment (CosminPerRam) [reply lib/QueryRunner.js]: great catch, thanks.

**Review by CosminPerRam — APPROVED** [review]
